Do-It/Controller/CreateDoItTableViewController.swift

```diff
@@ -121,26 +127,42 @@ class CreateDoItTableViewController: UITableViewController {
             }
             courseVC.delegate = self
             show(courseVC, sender: sender)
+        case .alertOption:
+            guard case (nil, let alertVC) = AlertDateTableViewController.instantiateFromStoryboard() else {
+                fatalError("Navigation controller should not be attached in AlertDateTableViewController.storyboard")
+            }
+            alertVC.delegate = self
+            show(alertVC, sender: sender)
         default:
             break
         }
     }
 
     @IBAction func save(_ sender: Any) {
-        var doIt = DoIt(course: course, dueDate: date, description: desc, name: name,
-                        priority: priority, kind: .homework)
-
-        switch inputMode {
-        case .addNewDoIt?:
-            delegate?.createDoItViewController(self, didSaveDoIt: doIt)
-        case .editDoIt(let doItToEdit)?:
-            doIt.identifier = doItToEdit.identifier
-            delegate?.createDoItViewController(self, didEditDoIt: doIt)
-        case .none:
-            fatalError("inputMode is not set!")
+        if course.name == "" || name == "" || alertString == "" {
+            let alertController = UIAlertController(title: "You're not done yet!",
+                                                    message: "Please fill in all the required fields before adding the Do-It", preferredStyle: UIAlertController.Style.alert)
+            alertController.addAction(UIAlertAction(title: "Ok", style: UIAlertAction.Style.default, handler: nil))
+            self.present(alertController, animated: true, completion: nil)
+        } else {
+            var doIt = DoIt(course: course, dueDate: date, description: desc,
+                            name: name, priority: priority, kind: .homework)
+            let alertDate = Calendar.current.date(byAdding: alertOption, to: date)
+            let notifManager = NotificationManager()
+            switch inputMode {
+            case .addNewDoIt?:
+                delegate?.createDoItViewController(self, didSaveDoIt: doIt)
+                notifManager.setTrigger(doIt, alertDate!)
+            case .editDoIt(let doItToEdit)?:
+                notifManager.cancelNotification(doIt)
+                notifManager.setTrigger(doIt, alertDate!)
```

Comment:
This cancellation + setting notification trigger again is why the `reschedule` function exists on NotificationManager.